src/main/java/org/mastodon/util/TreeUtils.java

```diff
@@ -0,0 +1,99 @@
+package org.mastodon.util;
+
+import java.util.Collection;
+
+import org.mastodon.collection.RefCollections;
+import org.mastodon.collection.RefList;
+import org.mastodon.collection.RefSet;
+import org.mastodon.graph.Edge;
+import org.mastodon.graph.Graph;
+import org.mastodon.graph.Vertex;
+import org.mastodon.graph.algorithm.traversal.InverseDepthFirstIterator;
+
+public class TreeUtils
+{
+
+	private TreeUtils()
+	{
+		// prevent instantiation of utility class
+	}
+
+	/**
+	 * This method finds a subset of the given {@code selectedVertices} that
+	 * contains only the roots of the subtrees that are selected. The order
+	 * of the returned list follows the order of the {@code roots} list and
+	 * the order of the outgoing edges of the graphs vertices.
+	 * <p>
+	 * Example:
+	 * <pre>
+	 *   A                B
+	 *  / \             /   \
+	 * a1  a2          b1    b2
+	 *     |          / \   / \
+	 *     a3        b3 b4 b5 b6
+	 *    / \
+	 *   a4 a5
+	 * </pre>
+	 *
+	 * If {@code selectedVertices} contains: {@code {a2, a4, a5, b1, b3, b4, b6}},
+	 * then the returned list will be: {@code [a2, b1, b6]}.
+	 */
+	public static < V extends Vertex< ? > > RefList< V > findSelectedSubtreeRoots(
+			final Graph< V, ? > graph,
+			final RefList< V > roots,
+			final RefSet< V > selectedVertices )
+	{
+		final RefList< V > selectedSubtreeRoots = RefCollections.createRefList( graph.vertices() );
+		final RefSet< V > visitedNodes = RefCollections.createRefSet( graph.vertices() );
+
+		for ( final V realRoot : roots )
```

Comment:
I am a bit confused now about what a `realRoot` is. There is now also a method in this in this class called `findRealRoots`. In this function a `realRoot` seems to be something else than within this method. If I understood this correctly, I would think the variable here should be named `root` instead of `realRoot`.